### PR TITLE
Update error-correct-pass2 and correct-reads without moving them from the sandbox

### PR DESCRIPTION
**Pull request by wltrimbl**
- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make clean diff-cover` If it introduces new functionality in
  `scripts/` is it tested?
- [x] `make format diff_pylint_report cppcheck doc pydocstyle` Is it well
  formatted?
- [x] Did it change the command-line interface? Only backwards-compatible
  additions are allowed without a major version increment. Changing file
  formats also requires a major version number increment.
- [x] For substantial changes or changes to the command-line interface, is it
  documented in `CHANGELOG.md`? See [keepachangelog](http://keepachangelog.com/)
  for more details.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)

### CODE REVIEW

**Comment by wltrimbl** [comment]
These changes do not introduce any new functionality; these merely make the two (wonderful) error-correction scripts work with khmer 2.1.2 and add tests that run most of the code included.  
Ready for review.

File: tests/test_sandbox_scripts.py [hunk 1]
```diff
@@ -341,3 +341,116 @@ def test_extract_compact_dbg_1():
...
+def test_correct_reads_duplicate():
+    script = 'correct-reads.py'
+    infile = utils.get_test_data('test-abund-read-paired.fa')
+    outfile = 'test-abund-read-paired.fa.corr'  # Warning output in cwd
+    args = []
+    args.extend([infile, infile])
+    try:
+        (status, out, err) = utils.runscript(script, args, sandbox=True)
+        assert 0  # assert raises error
+    except:
```
Comment (standage) [inline tests/test_sandbox_scripts.py]: Is this block trying to ensure the script fails on this data set? This is much more clearly declared with pytest's "raises" construct. Something like the following should work.

```python
with pytest.raises(ValueError) as ve:
    result = function_that_should_fail(bad_input)
assert 'expected error message' in str(ve)
```

If I've misunderstood, please clarify.

File: tests/test_sandbox_scripts.py [hunk 2]
```diff
@@ -341,3 +341,116 @@ def test_extract_compact_dbg_1():
...
+        pass
+
+
+@pytest.mark.skipif(not IN_REPOSITORY,
+                    reason='executing outside of the repository')
+def test_correct_reads_fq():
+    script = 'correct-reads.py'
+    infile = utils.get_test_data('test-reads.fq.bz2')
+    infile = utils.get_test_data('paired.fq')
```
Comment (standage) [inline tests/test_sandbox_scripts.py]: Looks like `infile` is redefined here.

File: sandbox/correct-reads.py [hunk 3]
```diff
@@ -330,43 +333,46 @@ def main():
...
+    print('wrote %d reads, %d bp' % (written_reads, written_bp,),
+          file=sys.stderr)
+    print('looked at %d reads twice (%.2f passes)' %
+          (save_pass2_total, n_passes), file=sys.stderr)
+    print('removed %d reads and corrected %d reads (%.2f%%)' %
+          (n_reads - written_reads, corrected_reads, percent_reads_corrected),
+          file=sys.stderr)
+    print('removed %.2f%% of bases (%d total)' %
+          ((1 - (written_bp / float(n_bp))) * 100.0, n_bp - written_bp),
+          file=sys.stderr)
```
Comment (standage) [inline sandbox/correct-reads.py]: Bless you for the tedious and thankless work towards Python 2/3 compatibility! 👍

**Review by standage — CHANGES_REQUESTED** [review]
Thanks for the PR!

This looks good for the most part. I've made a comments and questions, but these relate to minor issues that should be simple to address / comment on. I'll be happy to merge after your response.

**Review by standage — APPROVED** [review]

**Comment by standage** [comment]
Thanks for the updates. Will merge when the CI build passes.

**Comment by standage** [comment]
Thanks, @wltrimbl!

**Comment by ctb** [comment]
yes, thank you!!

**Pull request merged**